### PR TITLE
[path_provider] Restore 2.8 compatibility on Android and iOS

### PR DESCRIPTION
**Pull request by stuartmorgan**
Adds a new repo tooling command that removes dev_dependencies, which
aren't needed to consume a package, only for development. Also adds
a `--lib-only` flag to `analyze` to analyze only the client-facing code.
This is intended for use in the legacy analyze CI steps, primarily to
solve the problem that currently plugins that use Pigeon can't support a
version of Flutter older than the version supported by Pigeon, because
otherwise the legacy analysis CI steps fail.

Adds this new command to the legacy analysis CI step, and restores
the recently-removed 2.8/2.10 compatibility to path_provider.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
I forgot the pigeon file itself imports from `pigeon`; I think I'll have to add an option to restrict analysis to `lib/` as well. Back to the drawing board.

**Comment by gaaclarke** [comment]
We also talked about in the past about removing the Flutter dependency on Pigeon.  You didn't want to do that?  I know that wasn't a perfect solution since certain usage of pigeon requires certain versions of Flutter.

**Comment by stuartmorgan** [comment]
> We also talked about in the past about removing the Flutter dependency on Pigeon. You didn't want to do that? I know that wasn't a perfect solution since certain usage of pigeon requires certain versions of Flutter.

Removing the Flutter dependency won't change anything here; the minimum Dart version for Pigeon will still be the same.

**Comment by stuartmorgan** [comment]
Updated to remove all `dev_dependencies`, and to only analyze the `lib/` directory in legacy versions.

File: script/tool/lib/src/remove_dev_dependencies.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:yaml/yaml.dart';
+import 'package:yaml_edit/yaml_edit.dart';
+
+import 'common/package_looping_command.dart';
+import 'common/repository_package.dart';
+
+/// A command to remove dev_dependencies, wich are not used by package clients.
```
Comment (gaaclarke) [inline script/tool/lib/src/remove_dev_dependencies.dart:12]:
```suggestion
/// A command to remove dev_dependencies, which are not used by package clients.
```

**Review by gaaclarke — APPROVED** [review]

**Comment by stuartmorgan** [comment]
Landing manually since `submit-queue` is stale.

**Pull request merged**